pkg/p2p/kadcast/utils.go

```diff
@@ -164,7 +173,27 @@ func encodeRedPacket(byteNum uint16, peerAddr net.UDPAddr, payload []byte) []byt
 	return enc
 }
 
-// Decode a CircularQueue packet and return the
+// Encodes received TCP packets to send it through the
+// Ring to the packetProcess rutine.
+func encodeRedTCPPacket(byteNum uint16, peerAddr net.Addr, payload []byte) []byte {
+	peerDataStr := peerAddr.String()
+	encodedLen := len(payload) + 8
+	enc := make([]byte, encodedLen)
+	// Get numBytes as slice of bytes.
+	numBytes := getBytesFromUint16(byteNum)
+	// Append it to the resulting slice.
+	copy(enc[0:2], numBytes[0:2])
+	// Append Peer IP.
+	copy(enc[2:6], peerDataStr[0:4])
+	// Append Port
+	port := peerDataStr[4:6]
```

Comment:
What's the point of allocating this slice?

Reply:
Because copy will not accept the output of `getBytesFromUint16` since it's not indexable. 

So I need to allocate and then make the copy. 
And I'm doing the copy to avoid using `append` with variadics  for items which `len` is known.

Reply:
I'm talking specifically about `port`, which, one line down, you copy into `enc[6:8]`. You could do something like `copy(enc[6:8], peerDataStr[4:6])` and the comment will explain what it's doing, without having to allocate that other slice.

As a side note, if you `append` items into a slice without exceeding the `cap`, no re-allocations are made.

Reply:
I saw the port part now, I was looking another function on my code..

Also ye, I know the re-allocations, works as Vec in Rust, but Ema told me something about using append with variadics, and avoid it when possible if the len of the slices is known.

Idk, anyways, has been corrected! Thanks mate!